src/main/java/com/zendesk/maxwell/bootstrap/MaxwellBootstrapUtility.java

```diff
@@ -63,13 +51,52 @@ public void run() {
 		}
 	}
 
-	private int getInsertedRowsCount(Connection connection, long rowId) throws SQLException {
+
+	private void monitorProgress(Connection connection, Long rowId) throws SQLException {
+		addMonitorShutdownHook(rowId);
+
+		long rowCount = getTotalRowCount(connection, rowId);
+		long initialRowCount, insertedRowsCount;
+		initialRowCount = getInsertedRowsCount(connection, rowId);
+		Long startedTimeMillis = null;
+
+		insertedRowsCount = initialRowCount;
+		while ( !isComplete ) {
+			if ( insertedRowsCount < rowCount ) {
+				if ( startedTimeMillis == null && insertedRowsCount > 0 ) {
+					startedTimeMillis = System.currentTimeMillis();
+				}
+				insertedRowsCount = getInsertedRowsCount(connection, rowId);
+			}
+			isComplete = getIsComplete(connection, rowId);
```

Comment:
yeah, previously it threw a weird `SQLException`, now it throws something quite a bit nicer.